clang/lib/Driver/Driver.cpp

```diff
@@ -7244,6 +7334,22 @@ void Driver::BuildActions(Compilation &C, DerivedArgList &Args,
 
   handleArguments(C, Args, Inputs, Actions);
 
+  bool HasValidSYCLRuntime =
+      C.getInputArgs().hasFlag(options::OPT_fsycl, options::OPT_fno_sycl,
+                               false) ||
+      hasSYCLDeviceOnly(C.getInputArgs());
+  bool IsSYCLOffloadArchEnabled =
+      HasValidSYCLRuntime &&
+      C.getInputArgs().hasArg(options::OPT_offload_arch_EQ);
```

Comment:
This can be handled when evaluating the ToolChains to be added, effectively reducing the number of times we have to do the `-fsycl --offload-arch` logic.